apps/src/clientApi.js

```diff
@@ -123,17 +129,36 @@ ApiRequestHelper.prototype.postToGet = function (localUrl, data, callback) {
 ApiRequestHelper.prototype.delete = function (localUrl, callback) {
   $.ajax({
     url: this.apiBaseUrl_ + localUrl,
-    type: 'delete'
-  }).done(function (/*data, textStatus, jqXHR*/) {
-    callback(null, null);
-  }).fail(function (jqXHR, textStatus, errorThrown) {
-    callback(
-        new Error('textStatus: ' + textStatus + '; errorThrown: ' + errorThrown),
-        null);
+    type: 'delete',
+    success: function () {
+      callback(null, null);
+    },
+    error: function (jqXHR, textStatus, errorThrown) {
+      callback(
+          new Error('textStatus: ' + textStatus + '; errorThrown: ' + errorThrown),
+          null);
+    }
   });
 };
 
 /**
+ * @param {!string} localUrl
+ * @returns {Error|null} an error if the request fails
+ */
+ApiRequestHelper.prototype.synchronousDelete = function (localUrl) {
+  var error = null;
+  $.ajax({
+    url: this.apiBaseUrl_ + localUrl,
+    type: 'delete',
+    async: false,
+    error: function (jqXHR, textStatus, errorThrown) {
+      error = new Error('textStatus: ' + textStatus + '; errorThrown: ' + errorThrown);
+    }
+  });
+  return error;
```

Comment:
Right; I'm _only_ using a synchronous request in this case because we're trying to perform cleanup in response to the document's `unload` event.  Regular async requests usually fail in this case because the page finishes unloading before the request is complete.  [This page has some more information about AJAX on `unload`](http://www.hunlock.com/blogs/Mastering_The_Back_Button_With_Javascript).